packages/ng/src/app/select/select.popover.component.ts

```diff
@@ -0,0 +1,292 @@
+		});
+	}
+	/**
+	 * Search for highliting the option corresponding to the clue
+	 * @param clue
+	 */
+	search(clue: string = ''): void {
+		this.luOptions$.subscribe(selectOptions => {
+			this._highlightIndex = selectOptions.findIndex((selectOption) => {
+				return selectOption.viewValue === clue;
+			});
+
+			this._highlightIndex$.next(this._highlightIndex);
+		});
+	}
+	/**
+	 * Select the option (value) of the popover.
+	 * This method will fire an event of itemSelected
+	 * @param option : the option should be in the list of options of the popover
+	 */
+	selectOption(option: T): void {
+		this.luOptions$.subscribe(selectOptions => {
+			this._highlightIndex = selectOptions.findIndex((selectOption) => {
+				return this.same(selectOption.value, option);
+			});
+
+			this._highlightIndex$.next(this._highlightIndex);
+			this.itemSelected.emit(selectOptions[this._highlightIndex]);
+		});
+	}
+
+	// Comes from LuPopoverComponent
+	onMouseDown($e) {
+		$e.preventDefault();
+	}
+	/** Call when the key "Enter" is hit */
+	onEnterKeydown(): void {
+		this.selectOption(this._highlightedOption);
+	}
+	/**
+	 * Call when the key "Down" is hit
+	 * @param popoverOpen true if the popover is open
+	 */
+	onDownKeydown(popoverOpen: boolean = false): void {
+		this._highlightIndex++;
+		this._applyHighlightIndex();
+		if (!popoverOpen) {
+			this._selectHighlightOption();
+		}
+	}
+	/**
+	 * Call when the key "Up" is hit
+	 * @param popoverOpen true if the popover is open
+	 */
+	onUpKeydown(popoverOpen: boolean = false): void {
+		this._highlightIndex--;
+		this._applyHighlightIndex();
+		if (!popoverOpen) {
+			this._selectHighlightOption();
+		}
+	}
+	/**
+	 * Call when the key "Home" is hit
+	 * @param popoverOpen true if the popover is open
+	 */
+	onHomeKeydown(popoverOpen: boolean = false): void {
+		this._highlightIndex = 0;
+		this._applyHighlightIndex();
+		if (!popoverOpen) {
+			this._selectHighlightOption();
+		}
+	}
+	/**
+	 * Call when the key "End" is hit
+	 * @param popoverOpen true if the popover is open
+	 */
+	onEndKeydown(popoverOpen: boolean = false): void {
+		this._highlightIndex = this._optionsLength - 1;
+		this._applyHighlightIndex();
+		if (!popoverOpen) {
+			this._selectHighlightOption();
+		}
+	}
+	/**
+	 * Helper method for updating the highlight index (observable, ...)
+	 */
+	private _applyHighlightIndex(): void {
+		this._highlightIndex = Math.min(this._highlightIndex, this._optionsLength - 1);
+		this._highlightIndex = Math.max(this._highlightIndex, 0);
+		this._highlightIndex$.next(this._highlightIndex);
+	}
+
+	/**
+	 * Helper method to fire the event of selection
+	 */
+	private _selectHighlightOption(): void {
+		this.luOptions$.subscribe(selectOptions => {
+			if (!!selectOptions) {
+				this.itemSelected.emit(selectOptions[this._highlightIndex]);
+			}
+		});
+	}
+
+	/**
+	 * @returns an observable of the values of options
+	 */
+	protected getOptions(): Observable<T[]> {
+		return this._options$;
+	}
+}
```

Comment:
same remark as with api-directive. a lot of this code seems to be common between this and the api-picker.popover, is it planned to have some class extension shenanigans going ?

Reply:
same response as the WIP for the futur "luAutocomplete" I'm not sure about the name ;)